plugin/trino-mysql/src/test/java/io/trino/plugin/mysql/BaseMySqlIntegrationSmokeTest.java

```diff
@@ -421,21 +420,15 @@ public void testPredicatePushdown()
     }
 
     private AutoCloseable withTable(String tableName, String tableDefinition)
-            throws Exception
     {
-        execute(format("CREATE TABLE %s%s", tableName, tableDefinition));
+        mysqlServer.execute(format("CREATE TABLE %s%s", tableName, tableDefinition));
         return () -> {
             try {
-                execute(format("DROP TABLE %s", tableName));
+                mysqlServer.execute(format("DROP TABLE %s", tableName));
             }
             catch (RuntimeException e) {
                 throw new RuntimeException(e);
             }
         };
     }
-
-    private void execute(String sql)
-    {
-        mysqlServer.execute(sql, mysqlServer.getUsername(), mysqlServer.getPassword());
```

Comment:
This is wrong direction IMO.
Subclass `TestGlobalTransactionMySqlIntegrationSmokeTest` initializes the server on its own.
Instead of this, we should remove the server field in this class.

please drop this commit for now

Reply:
Notice that parameters `mysqlServer.getUsername(), mysqlServer.getPassword()` are useless as these are defaults. Then method exists only for `mysqlServer.execute(sql);`. Inlining this makes it easier. 

> Instead of this, we should remove the server field in this class.

I agree that this would be better. However, it is common that we inject infra for tests like that.

Do you suggest that we should make `execute` abstract? That would be on the other hand increase boilerplate on implementation side.

Reply:
> Do you suggest that we should make execute abstract? 

Yes, i think so, but let's out of scope here